inventory/group_vars/all.yml

```diff
@@ -300,9 +300,16 @@ nginx_sites:
 
       location / {
         {{ nginx_valid_methods }}
-        return 301 https://$host$request_uri;
+        return 301 https://{{ domain }}$request_uri;
       }
 
+  ofn_www:
+    - |
+      listen 443 ssl http2;
+      listen [::]:443 ssl http2;
+      server_name www.{{ domain }}
+      return 301 https://{{ domain }}$request_uri;
```

Comment:
How can people access this if we don't have the `www.` domain in the certbot list? :thinking: 
Do we need to add that? Or is it optional? Does that mean that new instances won't have `www` by default?

Reply:
> Does that mean that new instances won't have www by default?

`www` will redirect to the bare domain on all instances (new or old) :+1:

> How can people access this if we don't have the `www.` domain in the certbot list?

The `domain` variable in ofn-install will always be the bare domain from now on. Any incoming requests to `www.{{ domain }}` will be processed by this server block (by nginx) and be redirected to the bare domain (which will then be processed by the regular `ofn_443` server block).

If you're doing a simple 301 response you don't need to handle TLS termination, as it'll be handled in the subsequent URL, so any additional certificates for the www subdomain are effectively obsolete/ignored after this.

There would have been a bit less complexity if it weren't for Katuma, which uses `app.katuma.org` for their primary doamin in production. In that case, requests to `app.katuma.org` will be processed the same as before. Presumably the server won't ever receive requests on `www.app.katuma.org`, but if it does they should be redirected to `app.katuma.org` correctly.

Reply:
I'm still concerned about this one. Here's the scenario I'm thinking of:

- We deploy to au-prod.
- We have certificates for www and bare. All good.
- We don't renew the cert for www and it will expire.
- Someone enters www.openfoodnetwork.org.au in the browser. The default is http, the redirect works, all good.
- Someone links to https://www.openfoodnetwork.org.au and the browser can't establish a TLS connection to receive a redirect. :bomb: 
- Someone types the www URL in the browser and has something like a HttpsEverywhere plugin. :bomb: 
- Browser decide to make HTTPS the default. :bomb: 

You said that the redirect can be handled without TLS. But the above is in a TLS block and isn't the point of the encryption to hide and verify the request and response? So how can the server respond with a redirect if the TLS connection isn't established? Is that a fancy HTTP2 feature?

Reply:
Aha... yeah you're right, the TLS termination has to happen before the redirect can be made. In that case, we can keep the `www.` domain in the list of `certbot` domains, but it has to be lower down the list than the bare domain. I'll re-add that now and do some tests.

Reply:
I've updated it :+1: